pkg/client/create_builder.go

```diff
@@ -212,20 +219,75 @@ func (c *Client) fetchLifecycle(ctx context.Context, config pubbldr.LifecycleCon
 	var uri string
 	var err error
 	switch {
+	case isDockerURI(config.URI):
```

Comment:
I think you can use the following [function](https://github.com/buildpacks/pack/blob/main/pkg/buildpack/locator_type.go#L84-L86) that is doing the same thing